src/minreg.rs

```diff
@@ -0,0 +1,152 @@
+use crate::traits::{CmRDT, CvRDT};
+use serde::{Deserialize, Serialize};
+use std::{error, fmt};
+
+/// `MinReg` Holds a monotonically decreasing value that implements the Ord trait. For use of floating-point values,
+/// you must create a wrapper (or use a crate like `float-ord`).
+/// For modelling as a `CvRDT`:
+/// ```rust
+/// use crdts::{CvRDT,MinReg}
+/// let mut a = MinReg{ 3 };
+/// let b = MinReg{ 2 };
+///
+/// a.merge(b);
+/// asserteq!(a.val, 2);
+/// ```
+/// and `CmRDT`:
+/// ```rust
+/// use crdts::{CmRDT, MinReg}
+/// let mut a = MinReg{ 3 };
+/// let b = MinReg{ 2 };
+/// a.apply(b); // MinReg is also an Op
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct MinReg<V: Ord + Copy> {
```

Comment:
Same comments from `MaxReg`
- no constraints on `V` on struct definition, i.e.  it should be `struct MinReg<V>{...}
- the Op should be the value type `V`
- no validation errors, just return Ok(()) for validation
- and conform to the informal "Register API"